File: malevich-web/src/app/_transfer/orderPublicDto.ts
import {OrderTypeDto} from "./orderTypeDto";
import {ArtworkStockDto} from "./artworkStockDto";

export class OrderPublicDto {
  id: string;
  type: OrderTypeDto;
  artworkStock: ArtworkStockDto;
  amount: number;
  isOwn: boolean;
}

File: malevich-server/src/main/java/io/malevich/server/transfer/OrderPublicDto.java
package io.malevich.server.transfer;

import com.fasterxml.jackson.annotation.JsonView;
import com.yinyang.core.server.core.dto.View;
import lombok.Getter;
import lombok.Setter;


@Getter
@Setter
public class OrderPublicDto {

    @JsonView({View.Admin.class, View.Gallery.class, View.Trader.class})
    private Long id;

    private OrderTypeDto type;

    private ArtworkStockDto artworkStock;

    private Double amount;

    private Boolean isOwn;

}
